src/amount.cpp

```diff
@@ -19,10 +19,10 @@ CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nSize)
 
 CAmount CFeeRate::GetFee(size_t nSize) const
 {
-    CAmount nFee = nSatoshisPerK*nSize / 1000;
+    CAmount nFee = nSatoshisPerK * nSize / 1000;
 
-    if (nFee == 0 && nSatoshisPerK > 0)
-        nFee = nSatoshisPerK;
+    if (nFee == 0 && nSize != 0 && nSatoshisPerK != 0)
+        nFee = CAmount(1);
```

Comment:
Ideally this code should be removed entirely, but I don't see that happening soon taking into account the limited effect and the substantial amount of refactoring in wallet code.